src/convert.cpp

```diff
@@ -12,20 +12,32 @@
 
 #include "src/convert.h"
 #include "src/log.h"
+#include "src/jsset.h"
+
+ddwaf_object* to_ddwaf_object(
+  ddwaf_object *object,
+  Napi::Env env,
+  Napi::Value val,
+  int depth,
+  bool lim,
+  bool ignoreToJSON,
+  JsSet jsSet
+);
 
 ddwaf_object* to_ddwaf_object_array(
```

Comment:
when converting an array, should we include the array in the `Set`?

Reply:
we could add a test with `array = [{array}]`

Reply:
got the same idea a few comments above, :hagrid:

Reply:
yes, 
```
const payload = []
payload.push(payload)
```
is recursive and possible. 

Implemented and tests added.